cluster_experiments/analytics/analysis_plan.py

```diff
@@ -0,0 +1,271 @@
+
+    Attributes
+    ----------
+    tests : List[HypothesisTest]
+        A list of HypothesisTest instances
+    variants : List[Variant]
+        A list of Variant instances
+    variant_col : str
+        name of the column with the experiment groups
+    alpha : float
+        significance level used to construct confidence intervals
+
+    Methods
+    -------
+    __init__(self, tests: List[HypothesisTest], variants: List[Variant]):
+        Initializes the AnalysisPlan with the provided list of hypothesis tests and variants.
+    _validate_inputs(tests: List[HypothesisTest], variants: List[Variant]):
+        Validates the inputs for the AnalysisPlan class.
+    """
+
+    def __init__(
+        self,
+        tests: List[HypothesisTest],
+        variants: List[Variant],
+        variant_col: str = "treatment",
+        alpha: float = 0.05,
+    ):
+        """
+        Parameters
+        ----------
+        tests : List[HypothesisTest]
+            A list of HypothesisTest instances
+        variants : List[Variant]
+            A list of Variant instances
+        variant_col : str
+            The name of the column containing the variant names.
+        alpha : float
+            significance level used to construct confidence intervals
+        """
+        self._validate_inputs(tests, variants, variant_col)
+        self.tests = tests
+        self.variants = variants
+        self.variant_col = variant_col
+        self.alpha = alpha
+
+    @staticmethod
+    def _validate_inputs(
+        tests: List[HypothesisTest], variants: List[Variant], variant_col: str
+    ):
+        """
+        Validates the inputs for the AnalysisPlan class.
+
+        Parameters
+        ----------
+        tests : List[HypothesisTest]
+            A list of HypothesisTest instances
+        variants : List[Variant]
+            A list of Variant instances
+        variant_col : str
+            The name of the column containing the variant names.
+
+        Raises
+        ------
+        TypeError
+            If tests is not a list of HypothesisTest instances or if variants is not a list of Variant instances.
+        ValueError
+            If tests or variants are empty lists.
+        """
+        if not isinstance(tests, list) or not all(
+            isinstance(test, HypothesisTest) for test in tests
+        ):
+            raise TypeError("Tests must be a list of HypothesisTest instances")
+        if not isinstance(variants, list) or not all(
+            isinstance(variant, Variant) for variant in variants
+        ):
+            raise TypeError("Variants must be a list of Variant instances")
+        if not isinstance(variant_col, str):
+            raise TypeError("Variant_col must be a string")
+        if not tests:
+            raise ValueError("Tests list cannot be empty")
+        if not variants:
+            raise ValueError("Variants list cannot be empty")
+
+    def analyze(
```

Comment:
let's think of ways of reducing responsibilities of this method. as far as I see it does the following:
* iterate in tests and each test it:
   * prepares cupac config
   * creates analysis object
   * prepares data for each dimension
   * calls analysis_inference and metric means
   * create hypothesistestresults object
   * it also iterates on variants

let's start emptying it out. I think we can move stuff to hypothesistest